cryptofeed/backends/elastic.py

```diff
@@ -0,0 +1,102 @@
+'''
+Copyright (C) 2017-2019  Bryant Moscon - bmoscon@gmail.com
+
+Please see the LICENSE file for the terms and conditions
+associated with this software.
+'''
+from decimal import Decimal
+import logging
+import json
+import itertools
+
+from cryptofeed.defines import BID, ASK
+from cryptofeed.backends._util import book_convert, book_delta_convert, book_flatten
+from cryptofeed.backends.http import HTTPCallback
+from cryptofeed.exceptions import UnsupportedType
+
+
+LOG = logging.getLogger('feedhandler')
+
+
+class ElasticCallback(HTTPCallback):
+    def __init__(self, addr: str, index=None, numeric_type=str, **kwargs):
+        super().__init__(addr, **kwargs)
+        self.addr = f"{addr}/{index}/{index}"
+        self.session = None
+        self.numeric_type = numeric_type
+
+
+class TradeElastic(ElasticCallback):
+    def __init__(self, *args, index='trades', **kwargs):
+        super().__init__(*args, index=index, **kwargs)
+
+    async def __call__(self, *, feed: str, pair: str, side: str, amount: Decimal, price: Decimal, order_id=None, timestamp=None):
+        if order_id is None:
+            order_id = 'None'
+
+        trade = {
+            'pair': pair,
+            'side': side,
+            'id': order_id,
+            'amount': self.numeric_type(amount),
+            'price': self.numeric_type(price),
+            'timestamp': timestamp
+        }
+        await self.write('POST', json.dumps(trade), headers={'content-type': 'application/json'})
+
+
+class FundingElastic(ElasticCallback):
+    def __init__(self, *args, index='funding', **kwargs):
```

Comment:
whats the reason for *args rather than just taking addr (which seems like the only thing we'd expect there since you're passing it to the super which only takes that argument? This comment applies for all the classes that inherit from ElasticCallback

Reply:
is it just so you can easily change the super class without having to modify all of the inheriting ones? I guess the tradeoff there would be readability of the inheriting class vs ease of modifying the super